### PR TITLE
feat: lazygit integration

### PR DESCRIPTION
**Pull request by horriblename**
https://github.com/NotAShelf/neovim-flake/issues/46

a simple lazygit thing using toggleterm, derived from their help file

### CODE REVIEW

File: modules/terminal/toggleterm/config.nix [hunk 1]
```diff
@@ -6,31 +6,47 @@
...
+      + optionalString cfg.lazygit.enable ''
+
+        local terminal = require 'toggleterm.terminal'
+        local lazygit = terminal.Terminal:new({
+          cmd = "lazygit",
+          direction = '${cfg.lazygit.direction}',
+          hidden = true,
+          on_open = function(term)
+            vim.cmd("startinsert!")
+            vim.keymap.set( 't', [[${toggleKey}]], function() term:toggle() end, {silent = true, noremap = true, buffer = term.bufnr})
```
Comment (NotAShelf) [inline modules/terminal/toggleterm/config.nix:45]: A separate dag here could be a better approach. Though, this is not exactly wrong, as it is still a part of toggleterm.

File: modules/terminal/toggleterm/toggleterm.nix [hunk 2]
```diff
@@ -17,5 +17,13 @@ with builtins; {
       default = false;
       description = "Enable winbar";
     };
+    lazygit = {
+      enable = mkEnableOption "Enable LazyGit integration";
+      direction = mkOption {
+        type = types.enum ["horizontal" "vertical" "tab" "float"];
+        default = "float";
+        description = "Direction of the terminal";
```
Comment (NotAShelf) [inline modules/terminal/toggleterm/toggleterm.nix:25]: Clarify the terminal is for lazygit, "Direction of the lazygit window" would be more clear.

File: modules/terminal/toggleterm/config.nix [hunk 3]
```diff
@@ -6,31 +6,47 @@
 with lib;
 with builtins; let
   cfg = config.vim.terminal.toggleterm;
+  toggleKey = "<c-t>";
 in {
   config = mkIf cfg.enable {
     vim.startPlugins = [
       "toggleterm-nvim"
     ];
```
Comment (NotAShelf) [inline modules/terminal/toggleterm/config.nix:14]: I believe lazygit requires, well, lazygit to be installed. Perhaps we can have a module that takes `nullOr package` as a type to optionally let the user bring their own lazygit.

**Review by NotAShelf — CHANGES_REQUESTED** [review]

File: modules/terminal/toggleterm/config.nix (hunk 1 above)
Comment (horriblename) [reply modules/terminal/toggleterm/config.nix]: better check this one, I barely understood what dag is

File: modules/terminal/toggleterm/config.nix (hunk 3 above)
Comment (NotAShelf) [reply modules/terminal/toggleterm/config.nix]: You can use lib.optionals to append the package to the vim.startPlugins, which takes an input from the flake or a nixpkgs package. See `pluginsTypes` in `lib/types/plugins.nix`.

Comment (NotAShelf) [reply modules/terminal/toggleterm/config.nix]:
```nix
vim.startPlugins = [
    "toggleterm-nvim"
] ++ lib.optionals (lazygit.package != null) [
    cfg.lazygit.package
];
```

out of my head, this feels like a sane approach

**Comment by NotAShelf** [comment]
LGTM! I'll update the docs and should be ready to merge afterwards.

**Comment by NotAShelf** [comment]
Sorry it took me too long, had to go outside and touch grass for a bit. The alejandra formatting should fix the CI fail.

P.S. The styling CI looks like it's failing due to some Github networking error, Alejandra returns a positive exit code and I think we are good for a merge.

Thanks for your contribution.

**Pull request merged**